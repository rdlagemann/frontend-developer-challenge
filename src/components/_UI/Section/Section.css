.Section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Section__title {
  font-size: 18px;
  text-align: center;
  margin-top: 48px;
  margin-bottom: 48px;
}
