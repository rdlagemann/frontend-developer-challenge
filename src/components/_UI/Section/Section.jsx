import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Section.css'

Section.propTypes = {
  useDefaultClassNames: PropTypes.bool,
  title: PropTypes.string
}

Section.defaultProps = {
  useDefaultClassNames: true,
  title: ''
}

export default function Section({
  title,
  className,
  children,
  useDefaultClassNames,
  ...props
}) {
  const cssClasses = cx({
    'Section container': useDefaultClassNames,
    className
  })

  return (
    <section className={cssClasses} {...props}>
      {!!title && (
        <h2 className="Section__title" data-test="title">
          {title}
        </h2>
      )}
      {children}
    </section>
  )
}
