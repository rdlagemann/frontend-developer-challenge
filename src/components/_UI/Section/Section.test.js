import React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'
import Section from './Section'

describe('Section', () => {
  it('renders', () => {
    const output = create(<Section />)
    expect(output).toMatchSnapshot()
  })

  it('should have a title', () => {
    const title = 'TÃ­tulo'
    const section = shallow(<Section title={title} />)
    expect(section.find('[data-test="title"]').text()).toBe(title)
  })
})
