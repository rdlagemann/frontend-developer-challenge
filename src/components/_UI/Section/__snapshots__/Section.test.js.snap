// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders 1`] = `
<section
  className="Section container"
/>
`;
